### PR TITLE
[mimalloc] Update port to version 1.6.3

### PR DESCRIPTION
**Pull request by meirbon**
**Describe the pull request**

- What does your PR fix? Fixes #
It updates the mimalloc port to version 1.6.3 (the latest).

- Which triplets are supported/not supported? Have you updated the CI baseline?
The same as before.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
I think so? I am open to feedback.

### CODE REVIEW

**Comment by LilyWangL** [comment]
@MeirBon Thanks for your PR. Now port mimalloc build failed on CI test, you can get more error log from this URL: https://dev.azure.com/vcpkg/public/_build/results?buildId=40188&view=artifacts&type=publishedArtifacts

File: ports/mimalloc/CONTROL [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Source: mimalloc
-Version: 1.6.1-1
+Version: 1.6.3
 Description: Compact general purpose allocator with excellent performance
 Homepage: https://github.com/microsoft/mimalloc
```
Comment (LilyWangL) [inline ports/mimalloc/CONTROL:4]: Could you add [Supports](https://github.com/microsoft/vcpkg/blob/e62d1361288e83eba786395b60361ab35ba83800/docs/maintainers/control-files.md#supports) for this port?

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: ports/mimalloc/CONTROL (hunk 1 above)
Comment (MeirBon) [reply ports/mimalloc/CONTROL:4]: Added supports for the platforms that are listed in mimalloc's readme.

File: ports/mimalloc/portfile.cmake [hunk 2]
```diff
@@ -16,45 +14,134 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
     override    MI_OVERRIDE
 )
```
Comment (LilyWangL) [inline ports/mimalloc/portfile.cmake:16]: The way to handle feature has been replaced by [vcpkg_check_features()](https://github.com/microsoft/vcpkg/blob/a3a6530631df905eb5c0e26d0b20d7d548e0c465/docs/maintainers/vcpkg_check_features.md), so you should remove these codes:
```
if ("asm" IN_LIST FEATURES)
    set(MI_SEE_ASM ON)
else ()
    set(MI_SEE_ASM OFF)
endif ()

if ("override" IN_LIST FEATURES)
    set(MI_OVERRIDE ON)
else ()
    set(MI_OVERRIDE OFF)
endif ()

if ("secure" IN_LIST FEATURES)
    set(MI_SECURE ON)
else ()
    set(MI_SECURE OFF)
endif ()
```

File: ports/mimalloc/portfile.cmake [hunk 3]
```diff
@@ -16,45 +14,134 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
...
     OPTIONS_RELEASE
-        -DMI_CHECK_FULL=OFF
+        -DMI_DEBUG_FULL=OFF
     OPTIONS
         -DMI_INTERPOSE=ON
         -DMI_USE_CXX=OFF
         -DMI_BUILD_TESTS=OFF
+        -DMI_BUILD_SHARED=ON
+        -DMI_BUILD_STATIC=ON
+        -DMI_OVERRIDE=${MI_OVERRIDE}
```
Comment (LilyWangL) [inline ports/mimalloc/portfile.cmake:48]: These codes shouldn't be added, already has `${FEATURE_OPTIONS}`.
```
        -DMI_OVERRIDE=${MI_OVERRIDE}
        -DMI_SECURE=${MI_SECURE}
        -DMI_SEE_ASM=${MI_SEE_ASM}
```

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: ports/mimalloc/portfile.cmake [hunk 4]
```diff
@@ -16,45 +14,134 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
...
 )
 
 vcpkg_install_cmake()
 
 vcpkg_copy_pdbs()
 
-file(GLOB lib_directories RELATIVE ${CURRENT_PACKAGES_DIR}/lib "${CURRENT_PACKAGES_DIR}/lib/${PORT}-*")
-list(GET lib_directories 0 lib_install_dir)
-vcpkg_fixup_cmake_targets(CONFIG_PATH lib/${lib_install_dir}/cmake)
+file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/lib)
```
Comment (LilyWangL) [inline ports/mimalloc/portfile.cmake:58]: Thanks for your PR. The way to handle `*.lib, *.dll, *.h` work fine, why do these change? They have been installed to right directory.

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: ports/mimalloc/portfile.cmake (hunk 4 above)
Comment (LilyWangL) [reply ports/mimalloc/portfile.cmake:40]: Ping @MeirBon for respoense.

Comment (MeirBon) [reply ports/mimalloc/portfile.cmake:40]: I could not get it to work when I updated mimalloc to 1.6.3. I believe, that's why I changed it, if there are any better ways then feel free to make changes or tell me what to change.

**Comment by BillyONeal** [comment]
Just tested locally that this PR change still fails x86-windows. Given that we don't have a passing CI run and and no response since July, I'm closing this due to inactivity. @MeirBon if you're still interested in this, please feel to open a PR with more-or-less identical changes when you have the x86-windows build working.

**Pull request closed without merging**